src/scripts/sync/sync-projects-per-target.ts

```diff
@@ -0,0 +1,56 @@
+import type { requestsManager } from 'snyk-request-manager';
+import * as debugLib from 'debug';
+
+import { getGithubReposDefaultBranch } from '../../lib/source-handlers/github';
+import { compareAndUpdateBranches } from '../../lib/project/compare-branches';
+import type {
+  SnykProject, Target,
+} from '../../lib/types';
+import {
+  SupportedIntegrationTypesUpdateProject,
+} from '../../lib/types';
+import { targetGenerators } from '../generate-imported-targets-from-snyk';
+const debug = debugLib('snyk:sync-projects-per-target');
+
+export function getBranchGenerator(
+  origin: SupportedIntegrationTypesUpdateProject,
+): (target: Target, host?: string | undefined) => Promise<string> {
+  const getDefaultBranchGenerators = {
+    [SupportedIntegrationTypesUpdateProject.GITHUB]: getGithubReposDefaultBranch,
+  };
+  return getDefaultBranchGenerators[origin];
+}
+
+export async function updateProjectPerTarget(
```

Comment:
can we call this updateProjectForTarget? perTarget makes me think of a loop and there is no loop in this function

Reply:
updated